xdsl/dialects/arm_neon.py

```diff
@@ -179,34 +181,40 @@ def assembly_line(self):
 
 
 @irdl_op_definition
-class DSSFMulVecScalarOp(ARMInstruction):
+class DSSFMulOp(ARMInstruction):
     """
-    Floating-point multiply (mixed: first source operand is a vector, second is a scalar. Destination is a vector)
-    This instruction multiplies each of the floating-point values in the first source operand by the
-    second source operand and writes the resulting values to the corresponding lanes of the destination.
+    Floating-point multiply. Different instruction types supported:
+
+    1. Vector:  multiplies corresponding floating-point values in the vectors in the two
+    source NEON registers, and writes the result vector to the destination.
+
+    See external [documentation](https://developer.arm.com/documentation/100069/0606/SIMD-Vector-Instructions/FMUL--vector-).
+
+    2. Mixed: (first source operand is a vector, second is a scalar. Destination is a vector)
+    This instruction multiplies each of the floating-point values in the first source
+    operand by the second source operand and writes the resulting values to the corresponding
+    lanes of the destination.
     Encoding: FMUL <Vd>.<T>, <Vn>.<T>, <Vm>.<idx>.
-    Vd, Vn, Vm specify the regs. The <T> specifier determines element arrangement (size and count).
-    The <idx> specifier determines the index of Vm at which the second source operand (scalar) can be found,
-    preceded by a size specifier.
 
     See external [documentation](https://developer.arm.com/documentation/ddi0602/2024-12/SIMD-FP-Instructions/FMUL--vector---Floating-point-multiply--vector--?lang=en#T_option__4).
     """
 
-    name = "arm_neon.dss.fmulvec"
     d = result_def(NEONRegisterType)
     s1 = operand_def(NEONRegisterType)
     s2 = operand_def(NEONRegisterType)
-    scalar_idx = attr_def(IntegerAttr[i8])
-    arrangement = attr_def(NeonArrangementAttr)
+    scalar_idx = opt_prop_def(IntegerAttr[i8])
+    arrangement = prop_def(NeonArrangementAttr)
 
-    assembly_format = "$s1 `,` $s2 `[` $scalar_idx `]` $arrangement attr-dict `:` `(` type($s1) `,` type($s2) `)` `->` type($d)"
+    assembly_format = "$s1 `,` $s2 (`[` $scalar_idx^ `]`)? $arrangement attr-dict `:` \
+    functional-type(operands, $d)"
```

Comment:
```suggestion
    assembly_format = (
      "$s1 `,` $s2 (`[` $scalar_idx^ `]`)? $arrangement attr-dict "
      "`:` functional-type(operands, $d)"
    )
```
Does this work? looks a bit cleaner to me if so.